extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -103,7 +131,8 @@ public void handle(RoutingContext event) {
                 processRequestedHeaders(response, requestedHeaders);
             }
 
-            boolean allowsOrigin = isConfiguredWithWildcard(corsConfig.origins) || corsConfig.origins.get().contains(origin);
+            boolean allowsOrigin = isConfiguredWithWildcard(corsConfig.origins) || corsConfig.origins.get().contains(origin)
```

Comment:
You already know `corsConfig.origins.get()` will pass, so just pass that in and have `isOriginAllowedByRegex` accept a `List<String>`.

Reply:
That would make the function dependent on the order in which it's invoked.